File: algorithms/leetcode/lc028_implement_str_str/LC028ImplementStrStr.Tests.cs
using Xunit;

namespace AlgorithmDesign.leetcode.lc028_implement_str_str
{
    public class SolutionTests
    {
        [Fact]
        public void test_empty()
        {
            Assert.Equal(0, new Solution().StrStr("", ""));
        }

        [Fact]
        public void test_mississippi_a()
        {
            Assert.Equal(-1, new Solution().StrStr("mississippi", "a"));
        }

        [Fact]
        public void test_mississippi_si()
        {
            Assert.Equal(3, new Solution().StrStr("mississippi", "si"));
        }

        [Fact]
        public void test_bigger_in_smaller()
        {
            Assert.Equal(-1, new Solution().StrStr("", "test"));
        }
    }
}

File: algorithms/code/leetcode/lc028_implement_str_str/LC028ImplementStrStrTests.java
package leetcode.lc028_implement_str_str;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * https://leetcode.com/problems/implement-strstr/
 */
public final class LC028ImplementStrStrTests {
    @Test
    public void testEmpty() throws Exception {
        LC028ImplementStrStr solution;
        solution = new LC028ImplementStrStr();
        assertEquals(0, solution.strStr("", ""));
    }

    @Test
    public void testMississippiA() throws Exception {
        LC028ImplementStrStr solution;
        solution = new LC028ImplementStrStr();
        assertEquals(-1, solution.strStr("mississippi", "a"));
    }

    @Test
    public void testMississippiSi() throws Exception {
        LC028ImplementStrStr solution;
        solution = new LC028ImplementStrStr();
        assertEquals(3, solution.strStr("mississippi", "si"));
    }

    @Test
    public void testBiggerInSmaller() throws Exception {
        LC028ImplementStrStr solution;
        solution = new LC028ImplementStrStr();
        assertEquals(-1, solution.strStr("", "test"));
    }
}
